src/test/java/io/javarig/generation/javapredefinedtypes/StringGenerationTest.java

```diff
@@ -45,7 +49,38 @@ public void shouldGenerateStringWithSizeBetween() {
         log.info("shouldGenerateString : {}", generated);
         assertThat(generated).isNotNull();
         assertThat(generated).isInstanceOf(String.class);
-        assertThat(generated).asString().hasSizeBetween(minSize, maxSize);
+        assertThat(generated).asString().hasSizeBetween(minSize, maxSize-1);
+    }
+    @ParameterizedTest
+    @ValueSource(strings = {"", "[a-zA-Z0-9,;:!?]"})
+    public void shouldGenerateStringMatchingRegexPatternMatchingStringsSmallerThanGivenSize(String regexPattern) {
```

Comment:
should be shouldGenerateStringMatchingRegexPatternSmallerThanGivenSize ??

Reply:
when given a regex that cannot fulfill the size condition (for example given this regex "hello" with a max size of 2) we drop the size constraint

Reply:
The name is confusing. Don't you think "shouldPrioritizePattenConfigOverGivenSizeConfig" is better? And add the case of pattern size exceeding the given maxSize.

Reply:
Done